widgets/edd-widget-related-downloads.php

```diff
@@ -18,298 +17,298 @@
+						'numberposts'         => $limit,
+						'ignore_sticky_posts' => 1,
+						'post_status'         => 'publish',
+						'orderby'             => 'rand',
+					);
+
+					$related = get_posts( $params );
+
+					if ( ! empty( $related ) ) {
+						foreach ( $related as $related_download ) {
+							if ( $related_count === $limit ) {
+								break;
+							}
+							$related_downloads[ $related_download->ID ] = $related_download;
+						}
+					}
+				}
+
+				++$c;
+
+				// limit to 5 loops.
+				if ( $c > 5 ) {
+					break;
+				}
+			}
+
+			// return empty if there are no related downloads.
+			if ( empty( $related_downloads ) ) {
+				return;
+			}
+
+			// get the post type.
+			$post_type_obj = get_post_type_object( $post_type );
+
+			// start the list output.
+			$out .= "<ul class=\"widget-related-entries\">\n";
+
+			// set the link structure.
+			$link = "<a href=\"%s\" title=\"%s\" class=\"%s\" rel=\"bookmark\">%s</a>\n";
+
+			// filter the thumbnail size.
+			$thumbnail_size = apply_filters( 'edd_widgets_related_downloads_thumbnail_size', array( $thumbnail_size, $thumbnail_size ) );
+
+			// loop trough all downloads.
+			foreach ( $related_downloads as $download ) {
+				// get the title.
+				$title      = apply_filters( 'the_title', $download->post_title, $download->ID );
+				$title_attr = apply_filters( 'the_title_attribute', $download->post_title, $download->ID );
+
+				// get the post thumbnail.
+				if ( 1 === $thumbnail && has_post_thumbnail( $download->ID ) ) {
+					$post_thumbnail = get_the_post_thumbnail( $download->ID, $thumbnail_size, array( 'title' => esc_attr( $title_attr ) ) ) . "\n";
+					$out           .= "<li class=\"widget-download-with-thumbnail\">\n";
+					$out           .= sprintf( $link, get_permalink( $download->ID ), esc_attr( $title_attr ), 'widget-download-thumb', $post_thumbnail );
+				} else {
+					$out .= "<li>\n";
+				}
+
+				// append the download's title.
+				$out .= sprintf( $link, get_permalink( $download->ID ), esc_attr( $title_attr ), 'widget-download-title', $title );
+
+				// get the price.
+				if ( 1 === $show_price ) {
+					if ( edd_has_variable_prices( $download->ID ) ) {
+						$price = edd_price_range( $download->ID );
+					} else {
+						$price = edd_currency_filter( edd_get_download_price( $download->ID ) );
+					}
+					$out .= sprintf( "<span class=\"widget-download-price\">%s</span>\n", $price );
+				}
+
+				// finish this element.
+				$out .= "</li>\n";
+			}
+			// finish the list.
+			$out .= "</ul>\n";
+
+			echo $args['before_widget'] . $out . $args['after_widget'];
+		}
+
+
+		/**
+		 * Update
+		 *
+		 * @return   array
+		 * @since    1.0
+		 */
+		public function update( $new_instance, $old_instance ) {
+			$instance = $old_instance;
+
+			// sanitize title.
+			$instance['title'] = strip_tags( $new_instance['title'] );
+
+			// sanitize limit.
+			$instance['limit'] = isset( $instance['limit'] ) ? (int) $new_instance['limit'] : 4;
```

Comment:
```suggestion
			$instance['limit'] = isset( $new_instance['limit'] ) ? (int) $new_instance['limit'] : 4;
```
Make sure you are checking the new instance--